drivers/ethernet/phy/phy_ti_dp83825.c

```diff
@@ -0,0 +1,601 @@
+	if (ret) {
+		return ret;
+	}
+
+	/* Enable link state changed interrupt*/
+	reg_val |= PHY_TI_DP83825_MISR_REG_LSCE;
+
+	/* Write settings to misr register */
+	ret = phy_ti_dp83825_write(dev, PHY_TI_DP83825_MISR_REG, reg_val);
+	if (ret) {
+		return ret;
+	}
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+
+	return 0;
+}
+
+static int phy_ti_dp83825_reset(const struct device *dev)
+{
+	const struct ti_dp83825_config *config = dev->config;
+	struct ti_dp83825_data *data = dev->data;
+	int ret;
+
+	/* Lock mutex */
+	ret = k_mutex_lock(&data->mutex, K_FOREVER);
+	if (ret) {
+		LOG_ERR("PHY mutex lock error");
+		return ret;
+	}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	if (!config->reset_gpio.port) {
+		goto skip_reset_gpio;
+	}
+
+	/* Start reset (logically ACTIVE, physically LOW) */
+	ret = gpio_pin_set_dt(&config->reset_gpio, 1);
+	if (ret) {
+		goto done;
+	}
+
+	/* Reset pulse (minimum specified width is T1=25us) */
+	k_busy_wait(USEC_PER_MSEC * 1);
+
+	/* Reset over (logically INACTIVE, physically HIGH) */
+	ret = gpio_pin_set_dt(&config->reset_gpio, 0);
+
+	/* POR release time (minimum specified is T4=50ms) */
+	k_busy_wait(USEC_PER_MSEC * PHY_TI_DP83825_POR_DELAY);
+
+	goto done;
+skip_reset_gpio:
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios) */
+	ret = phy_ti_dp83825_write(dev, MII_BMCR, MII_BMCR_RESET);
+	if (ret) {
+		goto done;
+	}
+	/* POR release time (minimum specified is T4=50ms) */
+	k_busy_wait(USEC_PER_MSEC * PHY_TI_DP83825_POR_DELAY);
+
+done:
+	/* Unlock mutex */
+	k_mutex_unlock(&data->mutex);
+
+	LOG_DBG("PHY (%d) reset completed", config->addr);
+
+	return ret;
+}
+
+static int phy_ti_dp83825_cfg_link(const struct device *dev, enum phy_link_speed speeds)
+{
+	const struct ti_dp83825_config *config = dev->config;
+	struct ti_dp83825_data *data = dev->data;
+	int ret;
+	uint32_t anar;
+
+	/* Lock mutex */
+	ret = k_mutex_lock(&data->mutex, K_FOREVER);
+	if (ret) {
+		LOG_ERR("PHY mutex lock error");
+		goto done;
+	}
+
+	/* We are going to reconfigure the phy, don't need to monitor until done */
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+	if (!config->interrupt_gpio.port) {
+		k_work_cancel_delayable(&data->phy_monitor_work);
+	}
+#else
+	k_work_cancel_delayable(&data->phy_monitor_work);
+#endif /* DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios) */
+
+	/* Reset PHY */
+	ret = phy_ti_dp83825_reset(dev);
+	if (ret) {
+		goto done;
+	}
+
+	/* DT configurations */
+	ret = phy_ti_dp83825_static_cfg(dev);
+	if (ret) {
+		goto done;
+	}
+
+	/* Read ANAR register to write back */
+	ret = phy_ti_dp83825_read(dev, MII_ANAR, &anar);
+	if (ret) {
+		LOG_ERR("Error reading phy (%d) advertising register", config->addr);
+		goto done;
+	}
+
+	/* Setup advertising register */
+	if (speeds & LINK_FULL_100BASE_T) {
+		anar |= MII_ADVERTISE_100_FULL;
+	} else {
+		anar &= ~MII_ADVERTISE_100_FULL;
+	}
+	if (speeds & LINK_HALF_100BASE_T) {
```

Comment:
Please add empty line after `if {}` block here and in whole file.

Reply:
done